persistence/src/main/java/de/cyface/persistence/FileUtils.java

```diff
@@ -0,0 +1,215 @@
+package de.cyface.persistence;
+
+import android.content.Context;
+import android.util.Log;
+
+import java.io.BufferedInputStream;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.FileFilter;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * Methods to access and find data in the filesystem.
+ *
+ * @author Armin Schnabel
+ * @version 1.0.0
+ * @since 3.0.0
+ */
+public class FileUtils {
+    /**
+     * Returns the path of the parent directory containing all measurement relevant data.
+     * This directory is deleted when the app is uninstalled and can only be accessed by the app.
+     */
+    private String measurementsDirPath;
+    /**
+     * Returns the path of the parent directory containing all folders of open measurements.
+     * This directory is deleted when the app is uninstalled and can only be accessed by the app.
+     */
+    private String openMeasurementsDirPath;
+    /**
+     * Returns the path of the parent directory containing all folders of finished measurements.
+     * This directory is deleted when the app is uninstalled and can only be accessed by the app.
+     */
+    private String finishedMeasurementsDirPath;
+    /**
+     * Returns the path of the parent directory containing all folders of synchronized measurements.
+     * This directory is deleted when the app is uninstalled and can only be accessed by the app.
+     */
+    private String synchronizedMeasurementsDirPath;
+    /**
+     * Returns the path of the parent directory containing all folders of corrupted measurements.
+     * This directory is deleted when the app is uninstalled and can only be accessed by the app.
+     */
+    private String corruptedMeasurementsDirPath;
+
+    /**
+     * @param context The {@link Context} required to locate the app's internal storage directory.
+     */
+    public FileUtils(final Context context) {
```

Comment:
If you remove that context and add the files dir as a parameter you could test the whole class using simple unit tests, avoiding any Android shenanigans.

Reply:
Alread done. Had to google "shenanigans" though :jack_o_lantern: